official/resnet/cifar10_main.py

```diff
@@ -252,13 +252,17 @@ def run_cifar(flags_obj):
   Returns:
     Dictionary of results. Including final accuracy.
   """
+  resnet_run_loop.resolve_flag_conflict()
+
   if flags_obj.image_bytes_as_serving_input:
     tf.compat.v1.logging.fatal(
         '--image_bytes_as_serving_input cannot be set to True for CIFAR. '
         'This flag is only applicable to ImageNet.')
     return
 
-  input_function = (flags_obj.use_synthetic_data and
+  use_synthetic_data = (flags_obj.data_source_type == "synthetic_data")
```

Comment:
This is not backward compatible. See my comment near the resolve conflict method